### PR TITLE
Tasks in SD with "accepted=False" are marked as done in ALM during a sync

### PR DESCRIPTION
**Pull request by keyvanm**
Tasks in SD with "accepted=False" are marked as done in ALM during a sync

### CODE REVIEW

File: sdetools/alm_integration/alm_plugin_base.py [hunk 1]
```diff
@@ -796,6 +798,11 @@ def synchronize(self):
                     if not self.config['alm_standard_workflow']:
                         continue
 
+                    if not sde_task['accepted']:
```
Comment (funkaoshi) [inline sdetools/alm_integration/alm_plugin_base.py]: I'd add little comments about what's up. Eg. `# Close tasks that are no longer part of the SDE project.`

File: sdetools/sdelib/sdeapi.py [hunk 2]
```diff
@@ -134,7 +134,7 @@ def get_projects(self, application, options={}, **filters):
                 call_headers=_encode_options(options))
         return response['results']
 
-    def get_tasks(self, project_id, args=None, **filters):
+    def get_tasks(self, project_id, args={}, **filters):
```
Comment (funkaoshi) [inline sdetools/sdelib/sdeapi.py]: Why the change here? Normally you shouldn't make python default arguments mutable objects.

File: sdetools/alm_integration/alm_plugin_base.py [hunk 3]
```diff
@@ -764,13 +764,15 @@ def synchronize(self):
             progress += 2
             self.output_progress(progress)
 
-            # Attempt to get all tasks
+            # Attempt to get all (Accepted) tasks
             tasks = self.sde_get_tasks()
```
Comment (sibartlett) [inline sdetools/alm_integration/alm_plugin_base.py]: Rename this variable to `accepted_tasks`.

File: sdetools/alm_integration/alm_plugin_base.py [hunk 4]
```diff
@@ -764,13 +764,15 @@ def synchronize(self):
...
+            unaccepted_tasks = self.sde_get_tasks(accepted=False)
             logger.info('Retrieved all tasks from SDE')
 
             # Filter tasks - progress must match reality
             tasks = self.filter_tasks(tasks)
```
Comment (sibartlett) [inline sdetools/alm_integration/alm_plugin_base.py]: Then change these 2 lines to:

```
tasks = self.filter_tasks(accepted_tasks + unaccepted_tasks)
```

File: sdetools/sdelib/sdeapi.py (hunk 2 above)
Comment (keyvanm) [inline sdetools/sdelib/sdeapi.py]: If for some reason the arg is set to None, the first line of the method will throw an error.
```python
args.update({'include': 'tags,how_tos,problem,related', 'expand': 'status,phase', 'absolute_urls': True})
```

Comment (JeffBain) [inline sdetools/sdelib/sdeapi.py]: Seems like we need a

    if args is none:
        args = {}

**Comment by sibartlett** [comment]
Merged in #405.
